micrometer-spring-legacy/src/samples/java/io/micrometer/spring/samples/components/okhttp/OkHttpMicrometerInterceptor.java

```diff
@@ -0,0 +1,100 @@
+package io.micrometer.spring.samples.components.okhttp;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+public class OkHttpMicrometerInterceptor implements Interceptor {
+
+	public static final String MICROMETER_URI_HEADER = "MICROMETER_URI_HEADER";
+	private final String requestsMetricName;
+	private final boolean recordRequestPercentiles;
+	private final Function<Request, Optional<String>> urlMapper;
+	private final boolean emptyUriIfNoMatch;
+
+	private MeterRegistry meterRegistry;
+
+
+	public OkHttpMicrometerInterceptor(MeterRegistry meterRegistry, String requestsMetricName, boolean recordRequestPercentiles) {
```

Comment:
Can we provide a default 'recommended' metric base name of `okhttp.client`?  Having useful default names will help common dashboard creation/discussion (I'm guessing you were mimicking the RestTemplate metrics?)

Reply:
Also allow a `defaultTags` parameter in case the user wants to include extra tags

Reply:
defaultTags as a List of Tags would be useful.

Can add default name